### ISSUE TITLE
Indeterminate progress  bar doesn't work smoothly when there is background work

### ISSUE THREAD

**Issue by TMaster** [angular/material2#11930]
#### Bug, feature request, or proposal:
Bug

#### What is the expected behavior?
The indeterminate progress bar should work smoothly at all times. 

#### What is the current behavior?
When issuing a request and waiting for a response the progress bar work smoothly.
However when doing some work in the background (ngrx for me) the progress bar behaves strangely.

#### What are the steps to reproduce?
https://angular-material2-issue-1stmzx.stackblitz.io

#### What is the use-case or motivation for changing an existing behavior?
I want the progress bar to work smoothly, 

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
Angular 6
Material 6.3.0
Windows 7 / 10
Typescript: 2.7.2
All browswers

**Comment by josephperrott** [comment]
Closing as duplicate of #10234

**Comment by TMaster** [comment]
I'm sorry but I seemed to lost you guys.

My initial problem was that I could not sort tables that have pipes in the `<td>` element #11782.

So I did what @IsNull suggested and create a view model instead of the existing model I get from the server. 

Now you are saying that this is the cause for the strange behavior of the progress spinner.

I'm sorry but I really lost here guys. What should I do?

Thanks
